merge:  봉사 신청 API 구현

in-jun opened the pull request:
## #️연관된 이슈

#29 

## 작업 내용

봉사 신청 기능을 구현한다.

- `VolunteerApplication` 엔티티 생성
- 봉사 신청 API 구현(/volunteer/applications/{postId})
- 봉사 신청 서비스 로직 구현
- 봉사 신청 응답 DTO 생성

Commit:
feat: (#29) 봉사 신청 API 구현

봉사 신청 기능을 구현한다.

- `VolunteerApplication` 엔티티 생성
- 봉사 신청 API 구현(/volunteer/applications/{postId})
- 봉사 신청 서비스 로직 구현
- 봉사 신청 응답 DTO 생성

coehgns commented on src/main/kotlin/org/example/root_be/domain/applications/service/ApplyVolunteerService.kt:
```diff
@@ -0,0 +1,37 @@
+    @Transactional
+    fun execute(postId: Long): ApplyVolunteerResponse {
+        val user = userFacade.getCurrentUser()
+        val post = volunteerPostRepository.findById(postId)
+            .orElseThrow() { VolunteerPostNotFoundException }
```
`VolunteerFacade`가 구현되어 있어서 활용하면 좋을 것 같습니다!

in-jun replied:
굳굳 리뷰 감사합니당 ^^

in-jun replied:
[5442835](https://github.com/team-root/ROOT_BE/pull/31/commits/54428352bb607940b9babe9da1f4910efb96bcf6)

Commit:
refactor: (#29) 봉사 신청 서비스 로직 개선

Facade 패턴을 적용하여 봉사 신청 서비스 로직을 개선한다.
- 기존에 Repository를 직접 사용하던 부분을 Facade를 통해 처리하도록 변경
- 이를 통해 서비스 로직의 의존성을 줄이고, 코드의 가독성 및 유지보수성 향상
- `VolunteerPostRepository` 의존성 제거
- `VolunteerFacade` 의존성 추가

coehgns reviewed (approved):
LGTM
